python/GangaLHCb/__init__.py

```diff
@@ -84,28 +65,20 @@ def _store_root_version():
 def _store_dirac_environment():
     from GangaDirac.Lib.Utilities.DiracUtilities import write_env_cache, get_env
     platform = os.environ['CMTOPT']
-    diracversion = Ganga.Utility.Config.getConfig('LHCb')['diracVersion']
+    diracversion = Ganga.Utility.Config.getConfig('LHCb')['DiracVersion']
     fdir = os.path.join(os.path.expanduser("~/.cache/Ganga/GangaLHCb"), platform)
-    fname = os.path.join(fdir, defaultLHCbDirac)
+    fname = os.path.join(fdir, diracversion)
     if not os.path.exists(fname) or not os.path.getsize(fname):
-        logger.info("Storing new LHCbDirac environment (%s:%s)" % (str(defaultLHCbDirac), str(platform)))
-        cmd = 'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=defaultLHCbDirac)
-        env = execute(cmd)
+      cmd =  'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=diracversion)
+      env = execute(cmd)
+      if len(env)!=0:
         env = eval(env)
         write_env_cache(env, fname)
-    if not diracversion == defaultLHCbDirac:
-      fname = os.path.join(fdir, diracversion)
-      if not os.path.exists(fname) or not os.path.getsize(fname):
-        cmd =  'lb-run LHCBDIRAC {version} python -c "import os; print(dict(os.environ))"'.format(version=diracversion)
-        env = execute(cmd)
-        if len(env)!=0:
-          env = eval(env)
-          write_env_cache(env, fname)
-          logger.info("Storing new LHCbDirac environment (%s:%s)" % (str(diracversion), str(platform)))
-        else:
-          logger.warning('LHCbDirac version %s does not exist, using default version %s instead',diracversion,defaultLHCbDirac)
-          diracversion = defaultLHCbDirac
-          fname = os.path.join(fdir, defaultLHCbDirac)
+        logger.info("Storing new LHCbDirac environment (%s:%s)" % (str(diracversion), str(platform)))
+      else:
+        msg = 'LHCbDirac version {version} does not exist'.format(version=diracversion)
+        logger.error(msg)
+        sys.exit() 
```

Comment:
Doesn't it make sense to raise an exception here as it's being handled through the loading of the plugins?